### ISSUE TITLE
Input: onChange is triggered only after focus out

### ISSUE THREAD

**Issue by willianba** [SAP/ui5-webcomponents#5157]
**Describe the bug**
States are not being updated without focusing out first.

**Isolated Example**
Example: https://codesandbox.io/s/icy-platform-xn1vcu?file=/src/App.js

**To Reproduce**
1. Write something in the input
2. Click on "Update". This time it will work. The text will be rendered below the input field.
3. Change the value in the input
4. Click "Update" again. Now it won't work.
5. Click "Update" again or focus out of the field before clicking on "Update". It should work now.

**Expected behavior**
I write something in the input and click the button, the text should be rendered.
I must not need to focus out before doing anything.

**UI5 Web Components for React Information**
`@ui5/webcomponents` version: latest (1.3.1)
`@ui5/webcomponents-react` version: latest (0.133.0)
Operating System: MacOS
Browser: Safari

**Additional context**
I think this issue is happening on the `ui5-web-components` itself. I searched in their repo and saw some bugs related to "fields not being updated". Probably there are other components that rely on the same interface that is providing this bug as well 😔

**Comment by IlianaB** [comment]
Hello @SAP/ui5-webcomponents-topic-rl I am forwarding this issue to you, because it seems there is a problem, when you try to update the value of a Text, depending on the value of an Input. Looks like _handleNativeInputChange is called, but it calls with a delay _handleChange, so the change event is fired, after the call of the click event of the Text, so that the Text is not getting updated (as it still works with the old value of the Input).

Could you please have a look?

Regards,
Iliana

**Comment by henriquemattos** [comment]
Hi, team. This input onChange/Focus out thing is really misbehaving for the DatePicker (and possibly for the DateTimePicker but I will verify later). When I enter a date manually in DatePicker and tab out of the input, the `isValid` and `isInValidRange` uses the old `value` to validate and returns true even if the current value is not acceptable (like `29/05/2022aaa`.

Seems like we have enough reports: #5260, #5173, #5137 ... And they are a month old. Currently I can't rely on the `isValid` and `isInValidRange` of the DatePicker and have to do the validation by hands using `dateValue` and exploiting the non-public `getFormat().parse()` methods (which would be nice to have them exposed ☺️ ).

And if considering exposing the `getFormat` and `parse`, one of them is throwing JS error, which breaks any execution afterwards. Would be nice to have a console error and a return `null|undefined|false` and let the client decide how to handle it.

Thanks in advance,
Henrique
